src/Microsoft.AspNetCore.Authentication/Events/FailureContext.cs

```diff
@@ -9,10 +9,14 @@ namespace Microsoft.AspNetCore.Authentication
     /// <summary>
     /// Provides failure context information to handler providers.
     /// </summary>
-    public class FailureContext : BaseControlContext
+    public class FailureContext : HandleRequestContext<RemoteAuthenticationOptions>
```

Comment:
So this really is a RemoteFailureContext isn't it?  Given that its using RemoteAuthenticationOptions.